src/election_anomaly/user_interface/__init__.py

```diff
@@ -610,33 +643,37 @@ def new_datafile(session,munger,raw_path,project_root=None,juris=None,results_in
 	if not juris:
 		juris = pick_juris_from_filesystem(
 			project_root,juriss_dir='jurisdictions')
-	raw = read_datafile(munger,raw_path)
+	raw = read_combine_results(munger, raw_path, project_root,aux_data_dir=aux_data_dir)
 
 	if raw.empty:
+		# TODO improve error tracking
 		print('Datafile unable to be parsed with munger. Results not loaded to database. '
 			'Please check compatibilty between the two and try again.')
 		return
 	
 	count_columns_by_name = [raw.columns[x] for x in munger.count_columns]
 
 	try:
-		raw = mr.clean_raw_df(raw,munger)
+		raw = mr.munge_clean(raw, munger)
 	except:
-		print('Datafile unable to be parsed with munger. Results not loaded to database. '
+		print('Cleaning of datafile failed.. Results not loaded to database. '
 			'Please check compatibilty between the two and try again.')
 		return
 
 	# NB: info_cols will have suffix added by munger
 
 	# check jurisdiction against raw results file, adapting jurisdiction files as necessary
 	# TODO: incorporate juris.check_against_raw_results(raw,munger,count_columns_by_name)
-	# if jurisdction changed, load to db
+	# if jurisdiction changed, load to db
 	juris.load_juris_to_db(session,project_root)
 
 	try:
-		mr.raw_elements_to_cdf(session,project_root,juris,munger,raw,count_columns_by_name,results_info)
+		err = mr.raw_elements_to_cdf(session,project_root,juris,munger,raw,count_columns_by_name,results_info)
+		if err:
```

Comment:
Isn't this (and the next line) handled by the `except` block? I'm not sure why we return an error and print it but continue in the function as if this portion was successful.

Reply:
The intent is to give some nuance in the error messages. There are a variety of errors possible in  `raw_elements_to_cdf()`, which are described in `err` if they occur (and shown to the user). If `raw_elements_to_cdf()` fails in some other way, the user will get the more generic, less helpful,  'Datafile not loaded'.

The error-recording in raw_elements_to_cdf() was necessary for me to diagnose some problems in the MI munge (which turned out to be a consequence of duplicate Election_Id columns. I can probably get rid of some of these error messages.